### PR TITLE
[Refactor] Remove dataset/tokenizer/model packaging under common, make pretrain/finetune scripts without common dependency.

### PR DESCRIPTION
**Pull request by harborn**
Remove dataset/tokenizer/model packaging under common, make pretrain/finetune scripts without common dependency.

### CODE REVIEW

File: llm_on_ray/finetune/finetune.py [hunk 1]
```diff
@@ -190,6 +110,75 @@ def convert_dtype(dtype: str) -> Optional[torch.dtype]:
...
+            )
+            datasets_dict["validation"] = datasets_dict["test"]
+            return datasets_dict
+        return datasets.DatasetDict({"train": train_dataset})
+    else:
+        load_config = config.get("load_config", {})
+        if load_from_disk:
+            return datasets.load_from_disk(name, **load_config)
+        else:
+            return datasets.load_dataset(name, **load_config)
```
Comment (minmingzhu) [inline llm_on_ray/finetune/finetune.py:152]: In my opinion, the validation_split_percentage parameter from the Hugging Face dataset library is a useful feature for splitting the validation dataset. During my development of DPO, I utilized this parameter to split the validation data from the original dataset. This method ensures a straightforward and consistent way to create validation datasets by specifying the percentage of data to be used for validation.
        if data_args.dataset_name is not None:
            # Downloading and loading a dataset from the hub.
            raw_datasets = load_dataset(
                data_args.dataset_name,
                data_args.dataset_config_name,
                cache_dir=model_args.cache_dir,
                token=model_args.token,
                streaming=data_args.streaming,
            )

            if "validation" not in raw_datasets.keys() and training_args.do_eval:
                raw_datasets["validation"] = load_dataset(
                    data_args.dataset_name,
                    data_args.dataset_config_name,
                    split=f"train[:{data_args.validation_split_percentage}%]",
                    cache_dir=model_args.cache_dir,
                    token=model_args.token,
                    streaming=data_args.streaming,
                )
                raw_datasets["train"] = load_dataset(
                    data_args.dataset_name,
                    data_args.dataset_config_name,
                    split=f"train[{data_args.validation_split_percentage}%:]",
                    cache_dir=model_args.cache_dir,
                    token=model_args.token,
                    streaming=data_args.streaming,
                )

File: docs/finetune_parameters.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ The following are the parameters supported in the finetuning workflow.
 |tokenizer_name|None|Path to pretrained tokenizer from huggingface.co/models. If not provided, the tokenizer will be loaded from the `base_model`.|
 |gpt_base_model|True|This parameter is for [Transformers#22482](https://github.com/huggingface/transformers/issues/22482). It needs to be set to True when the pretrained model is realted to gpt, otherwise it is False.|
 |output_dir|/tmp/llm-ray/output|The output directory to store the finetuned model.|
+|report_to|none|The list of integrations to report the results and logs to.|
```
Comment (carsonwang) [inline docs/finetune_parameters.md:13]: can you please list the supported options? Can you also show the picture for one of your runs?

File: llm_on_ray/finetune/finetune.py [hunk 3]
```diff
@@ -21,10 +21,15 @@
...
 
-import torch
+from itertools import chain
 
+import torch
+import datasets
 import transformers
 
+from peft import get_peft_model, LoraConfig
+import deltatuner
```
Comment (carsonwang) [inline llm_on_ray/finetune/finetune.py:31]: I think we can remove deltatuner. It is not maintained. But we can do in a separate PR.

File: docs/finetune_parameters.md (hunk 2 above)
Comment (minmingzhu) [reply docs/finetune_parameters.md:13]: In my opinion, Do we only need to support one?  Because each tool needs to be pip. 

image

**Review by carsonwang — APPROVED** [review]
